modules/contracts/apps/challenge/src/handlers/execute.rs

```diff
@@ -152,349 +163,228 @@ fn update_friends_for_challenge(
 ) -> AppResult {
     app.admin.assert_admin(deps.as_ref(), &info.sender)?;
     let challenge = CHALLENGE_LIST.load(deps.storage, challenge_id)?;
-    if challenge.status != ChallengeStatus::Active {
-        return Err(AppError::WrongChallengeStatus {});
-    }
+    // Validate friend addr and account ids
+    let friends_validated: Vec<(Addr, Friend<Addr>)> = friends
+        .iter()
+        .cloned()
+        .map(|human| human.check(deps.as_ref(), app))
+        .collect::<AbstractSdkResult<_>>()?;
+
+    let (voters_addrs, friends): (Vec<Addr>, Vec<Friend<Addr>>) =
+        friends_validated.into_iter().unzip();
+
+    let proposal_id = challenge.current_proposal_id;
+
+    SIMPLE_VOTING
```

Comment:
Because changing the friends during the voting process can mess up the state of the proposal? 

It's like asking 3 people for their opinion but before hearing all of them you swap out someone who doesn't share your view. 

So I think the set of friends should be fixes as long as there are any proposals open. Does that make sense?